File: singly_linked_list/recipe-02/c/test_list_reverse.c
#include "string_list.h"
#include <stdio.h>

void print(ItemType data) 
{
    printf("data: %s\n", data);
}

void test()
{
    SinglyLinkedList* words = NewList();
    ListAppend(words, "foo");
    ListAppend(words, "bar");
    ListAppend(words, "bim");
    ListAppend(words, "baz");
    ListAppend(words, "quux");

    printf("The origin list\n");
    ListTravel(words, &print);

    printf("Reverse the list\n");
    ListReverse(words);
    ListTravel(words, &print);

    FreeList(words);
}

int main()
{
    Init();

    test();

    return 0;
}

File: singly_linked_list/recipe-02/python/test_list_reverse.py
from singly_linked_list import SinglyLinkedList

N = 10

def test():
    numbers = SinglyLinkedList()
    for i in range(2, N+1):
        numbers.Append(i)

    for number in numbers:
        print(number, end=" ")
    print()

    numbers.Reverse();

    for number in numbers:
        print(number, end=" ")
    print()

if __name__ == "__main__":
    test()
